pkg/metrics/series.go

```diff
@@ -90,7 +90,11 @@ func (series Series) Marshal() ([]byte, error) {
 //FIXME(olivier): remove this as soon as the v1 API can handle `device` as a regular tag
 func populateDeviceField(series Series) {
 	for _, serie := range series {
-		filteredTags := serie.Tags[:0] // use the same underlying array
+		// make a copy of the tags array. Otherwise the underlying array won't have
+		// the device tag for the Nth iteration (N>1), and the device field will
+		// be lost
+		var filteredTags []string
```

Comment:
this probably has a perf impact (since this makes the function allocate a new array for every `series`). I checked quickly on a dogstatsd-busy node and it didn't seem to have such an impact there, but we should benchmark this in the future.

Possible perf improvements: allocate a new array only if a `device:` tag is found ; allocate the slice with the correct capacity (since we should know the desired capacity in advance)